include/shark/Algorithms/DirectSearch/Indicators/HypervolumeIndicator.h

```diff
@@ -29,128 +29,70 @@
  * along with Shark.  If not, see <http://www.gnu.org/licenses/>.
  *
  */
-#ifndef SHARK_ALGORITHMS_DIRECTSEARCH_HYPERVOLUMEINDICATOR_H
-#define SHARK_ALGORITHMS_DIRECTSEARCH_HYPERVOLUMEINDICATOR_H
+#ifndef SHARK_ALGORITHMS_DIRECTSEARCH_INDICATORS_HYPERVOLUMEINDICATOR_H
+#define SHARK_ALGORITHMS_DIRECTSEARCH_INDICATORS_HYPERVOLUMEINDICATOR_H
 
 #include <shark/Core/Exception.h>
 #include <shark/Core/OpenMP.h>
 #include <shark/Algorithms/DirectSearch/Operators/Hypervolume/HypervolumeCalculator.h>
-#include <shark/LinAlg/Base.h>
 
 #include <algorithm>
 #include <vector>
 
 namespace shark {
 
-/**
-*  \brief Calculates the hypervolume covered by a front of non-dominated points.
-*/
+///  \brief Calculates the hypervolume covered by a front of non-dominated points.
 struct HypervolumeIndicator {
 public:
 	
-	/**
-	* \brief Calculates the volume of a hyperfront using a reference point
-	*
-	* \param [in,out] extractor Extractor instance that maps elements of the front to \f$\mathbb{R}^d\f$.
-	* \param [in] front pareto front of points to calculate the hypervolume for. 
-	* \param [in] referencePoint reference for measuring the hypervolume
-	*/
-	template<typename Extractor, typename ParetoFrontType, typename VectorType>
-	double operator()( Extractor const& extractor, ParetoFrontType const& front, VectorType const& referencePoint) {
-		
-		if(front.empty()) return 0;
-
-		return (m_hv( boost::adaptors::transform(front, extractor), referencePoint) );
-	}
-
-	/**
-	* \brief Executes the algorithm and calls to an instance of HypervolumeCalculator.
-	*
-	* This version uses the reference point estimated by the last call to updateInternals.
-	*
-	* \param [in] extractor Extractor instance that maps elements of the front to \f$\mathbb{R}^d\f$.
-	* \param [in] front front of points to calculate the hypervolume for. 
-	*/
-	template<typename Extractor, typename ParetoFrontType>
-	double operator()( Extractor const& extractor, ParetoFrontType const& front) {
-		return (*this)( extractor, front, m_reference);
-	}
-		
-	/**
-	* \brief Determines the individual contributing the least to the front it belongs to.
-	*
-	* \param [in] extractor Maps the individuals to the objective space.
-	* \param [in] front The front of non-dominated individuals.
-	* \param [in] referencePoint reference for measuring the hypervolume
-	*/
-	template<typename Extractor, typename ParetoFrontType, typename VectorType>
-	std::size_t leastContributor( Extractor const& extractor, ParetoFrontType const& front, VectorType const& referencePoint)
+	/// \brief Determines the point contributing the least hypervolume to the overall front of points.
+	///
+	/// This version uses the reference point estimated by the last call to updateInternals.
+	///
+	/// \param [in] front pareto front of points
+	template<typename ParetoFrontType>
+	std::size_t leastContributor( ParetoFrontType const& front)
```

Comment:
unfortunately, that is not very helpful. right now the assumption is that the input is some kind of range of fitness values. this is a different range than the range of individuals and thus an iterator would be used to get the index of it